compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -853,6 +854,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let item_name = item_segment.ident;
         let result = self
             .resolve_fully_qualified_call(span, item_name, ty.normalized, qself.span, hir_id)
+            .and_then(|r| {
+                // lint bare trait if the method is found in the trait
+                if span.edition() >= Edition::Edition2021 {
```

Comment:
I don't really like that the error here got moved from astconv... See above comment about delaying a bug or stashing a diagnostic.